src/server/services/pipeline-in-progress/validateReadings.js

```diff
@@ -12,13 +12,17 @@ const { log } = require('../../log');
  * @param {Reading[]} arrayToValidate
  * @param {dict} conditionSet used to validate readings (minVal, maxVal, minDate, maxDate, threshold, maxError)
  * @param {string} meterIdentifier identifier of meter being checked
+ * @returns {object} { validReadings, invalidReadings, errMsg }
  */
 function validateReadings(arrayToValidate, conditionSet, meterIdentifier = undefined) {
 	/* tslint:disable:no-string-literal */
 	const { validDates, errMsg: errMsgDate } = checkDate(arrayToValidate, conditionSet['minDate'], conditionSet['maxDate'], conditionSet['maxError'] / 2, meterIdentifier);
```

Comment:
It seems you changed the check functions to also return invalid items but this and the following line seems not to accept the new returned value.